[personality-insights] Call to profile() failing in Google stack

The following simple Personality Insights Java program is failing with a JavaParseException in the Google stack.  The source file is here (renamed for attachment): [PersonalityInsightsExample.txt](https://github.com/watson-developer-cloud/java-sdk/files/291896/PersonalityInsightsExample.txt).  The input file with the profile data is here (renamed from .json for attachment): [p.txt](https://github.com/watson-developer-cloud/java-sdk/files/291865/p.txt).  And the output of the stack trace is here: [out.txt](https://github.com/watson-developer-cloud/java-sdk/files/291866/out.txt). I'm not sure what the issue is with this one; the JSON file appears to conform to the expected input for parsing an argument to the service (the same file works from the command line with cURL), and the code matches what I see in examples.

Testing with **java-sdk-2.10.0-jar-with-dependencies.jar**.

German, thanks a lot for taking the time to reply and to correct my Json parsing!  I'm reopening with a few follow-up issues.

**1**. The first code snippet you include doesn't compile for me.  I don't know what Content.java is, and apparently, neither does the compiler. :-) 

```
com\ibm\watson\developer_cloud\personality_insights\v2\PersonalityInsightsExample.java:52: error: cannot find symbol
        Content content = new Gson().fromJson(jReader, Content.java);
                                                              ^
  symbol:   variable java
  location: class Content
```

I played around some with the code, but I can't find a definition for Content.java, and I'm just not sure what it is.

**2.** The second code snippet builds and runs but has some problems.

**2.A**  First, it reports the following ParseExceptions on the dates:  [java-errors.txt](https://github.com/watson-developer-cloud/java-sdk/files/302772/java-errors.txt).

**2.B**  Second, the output is different in a few ways from what I get from the API when I test with the same file via cURL.  The cURL output is [curl.txt](https://github.com/watson-developer-cloud/java-sdk/files/302776/curl.txt); the Java output is [java.txt](https://github.com/watson-developer-cloud/java-sdk/files/302777/java.txt).

The first difference is that the Java output does not include any date/time information, which is obviously because of the parse errors.  The second difference is admittedly cosmetic, but the ordering of the fields is markedly different.  I know that this makes absolutely no difference programmatically, which is the primary use of the output.  But it's a little problematic for documenting things, and if there were any way to produce output that's at least closer to the default service API, that would be helpful.  Again, trivial, but I wanted to mention it.

Thanks again, German!

--- a/src/main/java/com/ibm/watson/developer_cloud/personality_insights/v2/model/ContentItem.java
+++ b/src/main/java/com/ibm/watson/developer_cloud/personality_insights/v2/model/ContentItem.java
@@ -15,7 +15,9 @@ package com.ibm.watson.developer_cloud.personality_insights.v2.model;
 
 import java.util.Date;
 
+import com.google.gson.annotations.JsonAdapter;
 import com.ibm.watson.developer_cloud.service.model.GenericModel;
+import com.ibm.watson.developer_cloud.util.LongToDateTypeAdapter;
 
 /**
  * The Content to be analyzed.
@@ -24,6 +26,8 @@ public class ContentItem extends GenericModel {
   private String charset;
   private String content;
   private String contenttype;
+  
+  @JsonAdapter(LongToDateTypeAdapter.class)
   private Date created;
   private Boolean forward;
   private String id;
@@ -31,6 +35,8 @@ public class ContentItem extends GenericModel {
   private String parentid;
   private Boolean reply;
   private String sourceid;
+  
+  @JsonAdapter(LongToDateTypeAdapter.class)
   private Date updated;
   private String userid;
 

--- a/src/test/java/com/ibm/watson/developer_cloud/personality_insights/v2/PersonalityInsightsIT.java
+++ b/src/test/java/com/ibm/watson/developer_cloud/personality_insights/v2/PersonalityInsightsIT.java
@@ -24,9 +24,11 @@ import org.junit.Before;
 import org.junit.Test;
 
 import com.ibm.watson.developer_cloud.WatsonServiceTest;
+import com.ibm.watson.developer_cloud.personality_insights.v2.model.Content;
 import com.ibm.watson.developer_cloud.personality_insights.v2.model.ContentItem;
 import com.ibm.watson.developer_cloud.personality_insights.v2.model.Profile;
 import com.ibm.watson.developer_cloud.personality_insights.v2.model.ProfileOptions;
+import com.ibm.watson.developer_cloud.util.GsonSingleton;
 
 /**
  * Personality Insights Integration Tests.
@@ -73,13 +75,13 @@ public class PersonalityInsightsIT extends WatsonServiceTest {
   }
 
   /**
-   * Gets the profile with content items.
+   * Gets the profile from a single content item
    *
-   * @return the profile with content items
+   * @return the profile from a single content item
    * @throws Exception the exception
    */
   @Test
-  public void getProfileWithContentItems() throws Exception {
+  public void getProfileWithASingleContentItem() throws Exception {
     File file = new File("src/test/resources/personality_insights/en.txt");
     String englishText = getStringFromInputStream(new FileInputStream(file));
 
@@ -91,4 +93,20 @@ public class PersonalityInsightsIT extends WatsonServiceTest {
   }
 
 
+  /**
+   * Gets the profile from a list of content items.
+   *
+   * @return the profile with content items
+   * @throws Exception the exception
+   */
+  @Test
+  public void getProfileWithContentItems() throws Exception {
+    File file = new File("src/test/resources/personality_insights/contentitems.json");
+    String contentItems = getStringFromInputStream(new FileInputStream(file));
+    Content content = GsonSingleton.getGson().fromJson(contentItems, Content.class);
+    ProfileOptions options = new ProfileOptions.Builder().contentItems(content.getContentItems()).build();
+    Profile profile = service.getProfile(options).execute();
+
+    assertProfile(profile);
+  }
 }

--- a/src/test/resources/personality_insights/contentItems.json
+++ b/src/test/resources/personality_insights/contentItems.json
@@ -1,0 +1,94 @@
+{
+   "contentItems": [
+      {
+         "content": "Wow, I liked @TheRock before , now I really SEE how special he is. The daughter story was IT for me. So great! #MasterClass",
+         "contenttype": "text/plain",
+         "created": 1447639154000,
+         "id": "666073008692314113",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      },
+      {
+         "content": ".@TheRock how did you Know to listen to your gut and Not go back to football? #Masterclass",
+         "contenttype": "text/plain",
+         "created": 1447638226000,
+         "id": "666069114889179136",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      },
+      {
+         "content": ".@TheRock moving back in with your parents so humbling. \" on the other side of your pain is something good if you can hold on\" #masterclass",
+         "contenttype": "text/plain",
+         "created": 1447638067000,
+         "id": "666068446325665792",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      },
+      {
+         "content": "Wow aren't you loving @TheRock and his candor? #Masterclass",
+         "contenttype": "text/plain",
+         "created": 1447637459000,
+         "id": "666065895932973057",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      },
+      {
+         "content": "RT @patt_t: @TheRock @Oprah @RichOnOWN @OWNTV this interview makes me like you as a fellow human even more for being so real.",
+         "contenttype": "text/plain",
+         "created": 1447637030000,
+         "id": "666064097562247168",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      },
+      {
+         "content": "\"Be You\".. That's the best advice ever @TheRock #MastersClass",
+         "contenttype": "text/plain",
+         "created": 1447636205000,
+         "id": "666060637181644800",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      },
+      {
+         "content": "Supersoulers let's lift our spirits pray and hold Paris in the Light\ud83d\ude4f\ud83c\udffe",
+         "contenttype": "text/plain",
+         "created": 1447602477000,
+         "id": "665919171062927360",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      },
+      {
+         "content": "RT @DeepakChopra: What I learned in week 1: Become What You Believe 21-Day Meditation Experience - https:\/\/t.co\/kqaMaMqEUp #GoogleAlerts",
+         "contenttype": "text/plain",
+         "created": 1447098990000,
+         "id": "663807393063538688",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      },
+      {
+         "content": "Watching Bryan Stevenson on #SuperSoulSunday! \"You are not the worst mistake you ever made\".\nAren't we glad  about that.",
+         "contenttype": "text/plain",
+         "created": 1446998643000,
+         "id": "663386507856736257",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      },
+      {
+         "content": ".@CherylStrayed  BRAVE ENOUGH my new favorite thing! Gonna buy a copy for all my girls. #Perfectgift https:\/\/t.co\/gz1tnv8t8K",
+         "contenttype": "text/plain",
+         "created": 1446915955000,
+         "id": "663039689360695296",
+         "language": "en",
+         "sourceid": "Twitter API",
+         "userid": "@Oprah"
+      }
+   ]
+}

